src/main/java/io/anserini/rerank/RerankerContext.java

```diff
@@ -27,16 +27,18 @@
   private final IndexSearcher searcher;
   private final Query query;
   private final K queryId;
+  private final String queryDocId; // this is for News Track Background Linking task
   private final String queryText;
   private final List<String> queryTokens;
   private final Query filter;
   private final SearchArgs searchArgs;
 
-  public RerankerContext(IndexSearcher searcher, K queryId, Query query, String queryText,
+  public RerankerContext(IndexSearcher searcher, K queryId, Query query, String queryDocId, String queryText,
       List<String> queryTokens, Query filter, SearchArgs searchArgs) throws IOException {
     this.searcher = searcher;
     this.query = query;
     this.queryId = queryId;
+    this.queryDocId = queryDocId;
```

Comment:
what exactly the is the queryDocId here?
We should be careful to add to `RerankerContext`... it's getting pretty bloated...

Reply:
Yes. 
So for News BL Track, the query is actually a docid and the task is to find the background news for that docid (news). There is also a query id from which the query docid is probably part of the ranking list.
It is probably wise to keep queryId, queryDocId and queryString at the same time.  
For example, the guideline explicitly states that duplicated results will be judged as non-relevant. So for the post-processing (implement as reranking) probably includes the comparison and dedup which requires the raw doc retrieved by queryDocId.